filestore/test/test_maps_reader.py

```diff
@@ -0,0 +1,191 @@
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)   #
+# HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,  #
+# STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OTHERWISE) ARISING   #
+# IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE   #
+# POSSIBILITY OF SUCH DAMAGE.                                          #
+########################################################################
+
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+
+import six
+from metadatastore.api import (insert_run_start,
+                               insert_event,
+                               insert_event_descriptor,
+                               insert_beamline_config)
+import filestore.retrieve as fsr
+import filestore.commands as fsc
+
+from filestore.file_readers import HDFMapsSpectrumHandler as HDFM
+
+import h5py
+import os.path as op
+import numpy as np
+from nose.tools import assert_true, assert_raises, assert_false, assert_equal
+import datetime
+import uuid
+
+import logging
+logger = logging.getLogger(__name__)
+
+
+fsr.register_handler('hdf_maps', HDFM)
+
+
+def save_syn_data(eid, data, base_path=None):
+    """
+    Save a array as hdf format to disk.
+    Defaults to saving files in :path:`~/.fs_cache/YYYY-MM-DD.h5`
+
+    Parameters
+    ----------
+    eid : unicode
+        id for file name
+    data : ndarray
+        The data to be saved
+    base_path : str, optional
+        The base-path to use for saving files.  If not given
+        default to `~/.fs_cache`.  Will add a sub-directory for
+        each day in this path.
+    """
+
+    if base_path is None:
+        base_path = op.join(op.expanduser('~'), '.fs_cache',
+                            str(datetime.date.today()))
+    fpath = op.join(base_path, str(eid) + '.h5')
+
+    with h5py.File(fpath, 'w') as f:
+        # create a group for maps to hold the data
+        mapsGrp = f.create_group('MAPS')
+        # now set a comment
+        mapsGrp.attrs['comments'] = 'MAPS group'
+
+        entryname = 'mca_arr'
+        comment = 'These are raw spectrum data.'
+        ds_data = mapsGrp.create_dataset(entryname, data=data)
+        ds_data.attrs['comments'] = comment
+    return fpath
+
+
+def get_data(ind_v, ind_h):
+    """
+    Get data for given x, y index.
+
+    Parameters
+    ----------
+    ind_v : int
+        vertical index
+    ind_h : int
+        horizontal index
+
+    Returns
+    -------
+    unicode:
+        id number of event
+    """
+
+    uid = str(uuid.uuid1())
+
+    # generate 3D random number with a given shape
+    syn_data = np.random.randn(20, 1, 10)
+    file_path = save_syn_data(uid, syn_data)
+
+    custom = {'dset_path': 'mca_arr'}
+
+    fb = fsc.save_file_base('hdf_maps', file_path, custom)
+    evl = fsc.save_file_event_link(fb, uid, link_parameters={'x': ind_v, 'y': ind_h})
+    return evl.event_id
+
+
+def test_data_io():
+    """
```

Comment:
I like what you're trying to do with the data io here, but since you're combining filestore and metadatastore, I would strongly advocate for this living in dataportal examples